### PR TITLE
to_nf4 and support for _to_copy

### PR DESCRIPTION
**Pull request by cpuhrsch**
Need test

Need ability to load into nf4 instance.

### CODE REVIEW

File: test/modules/test_nf4_linear.py [hunk 1]
```diff
@@ -170,5 +164,18 @@ def test_load_from_nf4_diff_meta(self):
         assert other_mod.param.block_size == 64
         assert other_mod.param.scaler_block_size == 1
 
+    def test_to_copy(self):
+        inpt_tensor = torch.rand(128, device='cpu')
```
Comment (rohan-varma) [inline test/modules/test_nf4_linear.py:168]: need to initialize it as bf16?

File: torchao/dtypes/nf4tensor.py [hunk 2]
```diff
@@ -501,3 +509,9 @@ def linear_nf4(input: torch.Tensor, weight: NF4Tensor) -> torch.Tensor:
         weight: NF4Tensor weight
     """
     return LinearNF4.apply(input, weight)
+
+def to_nf4(tensor,
+           block_size: int = 64,
+           scaler_block_size: int = 256):
+    tensor1 = tensor.to(torch.bfloat16)
```
Comment (rohan-varma) [inline torchao/dtypes/nf4tensor.py:516]: Would vote for raising error if input is not of bf16 dtype, as this downcast could be unexpected for users. Will let you and @drisspg make the call though

**Review by rohan-varma — APPROVED** [review]
LG, pls take a loook @drisspg

File: torchao/dtypes/nf4tensor.py (hunk 2 above)
Comment (drisspg) [reply torchao/dtypes/nf4tensor.py:516]: I think this makes sense, tbh, I don't think I made any assumptions in the code that require the original weigh to bfloat16, and this was motivated by checkpoint weights being in bf16.

Also I I think if this is the main entrypoint for users to create nf4 tensors then we should document this function

File: test/modules/test_nf4_linear.py [hunk 3]
```diff
@@ -170,5 +164,18 @@ def test_load_from_nf4_diff_meta(self):
...
+        inpt_tensor_nf4 = to_nf4(inpt_tensor, 32, 2)
+        inpt_tensor_bfloat16 = inpt_tensor_nf4.to(torch.bfloat16)
+        torch.testing.assert_allclose(inpt_tensor, inpt_tensor_bfloat16, atol=0.13, rtol=0.13)
+
+        if torch.cuda.is_available():
```
Comment (drisspg) [inline test/modules/test_nf4_linear.py:173]: nit: parametrize on test, but I arguably over-parametrize my tests

File: test/modules/test_nf4_linear.py (hunk 1 above)
Comment (cpuhrsch) [reply test/modules/test_nf4_linear.py:168]: I specifically wanted to try some other dtypes. To be rigorous I'd try them all, but that might not be worth the additional test power unless we do something special.

**Pull request merged**